src/cryptonote_basic/miner.cpp

```diff
@@ -575,7 +577,26 @@ namespace cryptonote
         continue; // if interrupted because stop called, loop should end ..
       }
 
-      bool on_ac_power = !on_battery_power();
+      boost::tribool battery_powered(on_battery_power());
+      if(indeterminate( battery_powered ))
+      {
+        if( ++battery_query_failure_count > 3 )
+        {
+          MERROR("on_battery_power() call failed 3 times, shutting down mining");
+          stop();
+          break;
+        }
```

Comment:
This seems like another reason to put inside of the function block - it can have its unique delay interval separate from the ones dealing with calculating idle times.

Although, is there any good reason to check multiple times?  @moneromooo-monero brought this up too. I am conflicted because I agree it seems like something that ought to be done, yet I cannot come up with a reasonable justification. I'd have to peak the kernel implementation to say more really ...